### PR TITLE
remove profit cap of limit order

### PR DESCRIPTION
**Pull request by charlesjhongc**
Remove profit cap and in fillByProtocol case.

### CODE REVIEW

File: contracts/test/forkMainnet/LimitOrder.t.sol [hunk 1]
```diff
@@ -1056,9 +1042,71 @@ contract LimitOrderTest is StrategySharedSetup {
...
+            relayer, // relayer
+            relayer, // profitRecipient
+            fillReceipt,
+            profit // takerTokenProfit
+        );
+        userProxy.toLimitOrder(_genFillByProtocolPayload(order, makerSig, protocolParams, crdParams));
+        vm.stopPrank();
+
+        // whole profit should be given to relayer
+        console.logUint(profit);
```
Comment (pilagod) [inline contracts/test/forkMainnet/LimitOrder.t.sol:1105]: This line could be removed

File: contracts/LimitOrder.sol [hunk 2]
```diff
@@ -390,31 +379,16 @@ contract LimitOrder is ILimitOrder, BaseLibEIP712, SignatureValidator, Reentranc
...
-
-        // Cap taker token profit
-        uint256 takerTokenProfitCap = _mulFactor(_settlement.takerTokenAmount, profitCapFactor);
-        uint256 takerTokenProfit = takerTokenExtra > takerTokenProfitCap ? takerTokenProfitCap : takerTokenExtra;
-
-        // Calculate taker token profit for relayer
-        uint256 takerTokenProfitFee = _mulFactor(takerTokenProfit, profitFeeFactor);
-        uint256 takerTokenProfitForRelayer = takerTokenProfit.sub(takerTokenProfitFee);
-
+        uint256 takerTokenProfit = takerTokenOut.sub(_settlement.takerTokenAmount);
```
Comment (NIC619) [inline contracts/LimitOrder.sol:382]: Maybe we can rename `takerTokenProfit` to `relayerTakerTokenProfit` or `takerTokenProfitForRelayer` along with the change?
To be more specific about where the profit should go.

File: contracts/test/forkMainnet/LimitOrder.t.sol (hunk 1 above)
Comment (charlesjhongc) [reply contracts/test/forkMainnet/LimitOrder.t.sol]: oops. Fixed in 5dacfe1

File: contracts/LimitOrder.sol (hunk 2 above)
Comment (charlesjhongc) [reply contracts/LimitOrder.sol]: Ok, renamed in 3c56458.

**Review by NIC619 — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

File: contracts/LimitOrder.sol [hunk 3]
```diff
@@ -435,13 +421,12 @@ contract LimitOrder is ILimitOrder, BaseLibEIP712, SignatureValidator, Reentranc
                 remainingAmount: _settlement.remainingAmount,
                 makerTokenFee: 0,
                 takerTokenFee: takerTokenFee,
-                takerTokenProfit: takerTokenProfit,
-                takerTokenProfitFee: takerTokenProfitFee,
-                takerTokenProfitBackToMaker: takerTokenProfitBackToMaker
+                relayerTakerTokenProfit: relayerTakerTokenProfit,
+                relayerProfitFee: relayerProfitFee
```
Comment (pilagod) [inline contracts/LimitOrder.sol:425]: Do you think it is worth to emphasize fee is in taker token like `relayerTakerTokenProfitFee`, which is also aligned with other variable naming.

File: contracts/test/forkMainnet/LimitOrder.t.sol [hunk 4]
```diff
@@ -1044,21 +1040,85 @@ contract LimitOrderTest is StrategySharedSetup {
...
+            ammOutputExtra.sub(relayerProfitFee), // relayerTakerTokenProfit
+            relayerProfitFee // relayerProfitFee
         );
         userProxy.toLimitOrder(payload);
         vm.stopPrank();
 
         userTakerAsset.assertChange(0);
-        // To avoid precision issue
-        receiverTakerAsset.assertChangeGt(int256(profit.mul(80).div(100)));
+        // To avoid precision issue, use great than instread
```
Comment (pilagod) [inline contracts/test/forkMainnet/LimitOrder.t.sol:1050]: typo: instead

File: contracts/test/forkMainnet/LimitOrder.t.sol [hunk 5]
```diff
@@ -1044,21 +1040,85 @@ contract LimitOrderTest is StrategySharedSetup {
...
+        receiverTakerAsset.assertChangeGt(int256(ammOutputExtra.mul(80).div(100)));
         makerTakerAsset.assertChange(int256(DEFAULT_ORDER.takerTokenAmount.mul(90).div(100)));
         makerMakerAsset.assertChange(-int256(DEFAULT_ORDER.makerTokenAmount));
-        uint256 feeTotal = DEFAULT_ORDER.takerTokenAmount.mul(10).div(100);
-        feeTotal = feeTotal.add(profit.mul(20).div(100));
-        fcTakerAsset.assertChange(int256(feeTotal));
+
+        // fee = taker fee + relayer profit fee
+        uint256 fee = DEFAULT_ORDER.takerTokenAmount.mul(10).div(100);
+        fee = fee.add(ammOutputExtra.mul(20).div(100));
```
Comment (pilagod) [inline contracts/test/forkMainnet/LimitOrder.t.sol:1057]: Maybe we could reuse the `relayerProfitFee` variable in line 1021.

File: contracts/LimitOrder.sol (hunk 3 above)
Comment (charlesjhongc) [reply contracts/LimitOrder.sol:425]: I thought about it but it may be a little bit lengthy. I'm ok with both. cc @NIC619

Comment (NIC619) [reply contracts/LimitOrder.sol:425]: Sounds good to me!

**Review by NIC619 — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Pull request merged**